robot-simulator/RobotSimulatorTest.swift

```diff
@@ -0,0 +1,150 @@
+        XCTAssertEqual(CardinalDirection.West, robot.bearing)
+    }
+    
+    func testTurnRightFromWest() {
+        robot.orient(.West)
+        robot.turnRight()
+        XCTAssertEqual(CardinalDirection.North, robot.bearing)
+    }
+    
+    func testTurnLeftFromNorth() {
+        robot.orient(.North)
+        robot.turnLeft()
+        XCTAssertEqual(CardinalDirection.West, robot.bearing)
+    }
+    
+    func testTurnLeftFromEast() {
+        robot.orient(.East)
+        robot.turnLeft()
+        XCTAssertEqual(CardinalDirection.North, robot.bearing)
+    }
+    
+    func testTurnLeftFromSouth() {
+        robot.orient(.South)
+        robot.turnLeft()
+        XCTAssertEqual(CardinalDirection.East, robot.bearing)
+    }
+    
+    func testTurnLeftFromWest() {
+        robot.orient(.West)
+        robot.turnLeft()
+        XCTAssertEqual(CardinalDirection.South, robot.bearing)
+    }
+
+    func testRobotCoordinates() {
+        robot.at(x: 3, y: 0)
+        XCTAssertEqual([3, 0], robot.coordinates)
+    }
+    
+    func testOtherRobotCoordinates() {
+        robot.at(x: -2, y: 5)
+        XCTAssertEqual([-2, 5], robot.coordinates)
+    }
+    
+    func testAdvanceWhenFacingNorth() {
+        robot.at(x: 0, y: 0)
+        robot.orient(.North)
+        robot.advance()
+        XCTAssertEqual([0, 1], robot.coordinates)
+    }
+    
+    func testAdvanceWhenFacingEast() {
+        robot.at(x: 0, y: 0)
+        robot.orient(.East)
+        robot.advance()
+        XCTAssertEqual([1, 0], robot.coordinates)
+    }
+    
+    func testAdvanceWhenFacingSouth() {
+        robot.at(x: 0, y: 0)
+        robot.orient(.South)
+        robot.advance()
+        XCTAssertEqual([0, -1], robot.coordinates)
+    }
+    
+    func testAdvanceWhenFacingWest() {
+        robot.at(x: 0, y: 0)
+        robot.orient(.West)
+        robot.advance()
+        XCTAssertEqual([-1, 0], robot.coordinates)
+    }
+    
+    func testInstructionForTurningLeft() {
+        XCTAssertEqual([.TurnLeft], simulator.instructions("L"))
+    }
+    
+    func testInstructionForTurningRight() {
+        XCTAssertEqual([.TurnRight], simulator.instructions("R"))
+    }
+    
+    func testInstructionForAdvancing() {
+        XCTAssertEqual([.Advance], simulator.instructions("A"))
+    }
+    
+    func testSeriesOfInstructions() {
+        XCTAssertEqual([.TurnRight, .Advance, .Advance, .TurnLeft], simulator.instructions("RAAL"))
+    }
+    
+    func testInstructRobot() {
+        let robot = SimulatedRobot()
+        simulator.place(robot, x: -2, y: 1, direction: .East)
+        simulator.evaluate(robot, commands: "RLAALAL")
+        XCTAssertEqual([0, 2], robot.coordinates)
+        XCTAssertEqual(CardinalDirection.West, robot.bearing)
+    }
+    
+    func testInstructManyRobots() {
+        let robot1 = SimulatedRobot()
+        let robot2 = SimulatedRobot()
+        let robot3 = SimulatedRobot()
+        simulator.place(robot1, x: 0, y: 0, direction: .North)
```

Comment:
But if we support `struct`s, wouldn't we also need to use `inout` params, so we're not just updating a copy of the robot?